src/Microsoft.NET.Sdk.Functions.Generator/FunctionJsonConverter.cs

```diff
@@ -173,36 +175,86 @@ private bool TryGenerateFunctionJsons()
-                        {
-                            var loadedModule = ModuleDefinition.ReadModule(tryPath, readerParams);
-                            exportedTypes = exportedTypes.Concat(loadedModule.Types);
-                        }
-                        catch (Exception ex)
+                        var tryPath = Path.Combine(assemblyRoot, $"{referencedAssembly.Name}.dll");
+                        if (File.Exists(tryPath))
                         {
-                            _logger.LogWarning($"Could not evaluate '{referencedAssembly.Name}' for function types. Exception message: {ex.Message}");
+                            try
+                            {
+                                var loadedModule = ModuleDefinition.ReadModule(tryPath, readerParams);
+                                exportedTypes = exportedTypes.Concat(loadedModule.Types);
+                            }
+                            catch (Exception ex)
+                            {
+                                _logger.LogWarning($"Could not evaluate '{referencedAssembly.Name}' for function types. Exception message: {ex.Message}");
+                            }
                         }
                     }
                 }
+
+                var functions = GenerateFunctions(exportedTypes).ToList();
+                foreach (var function in functions.Where(f => f.HasValue && !f.Value.outputFile.Exists).Select(f => f.Value))
+                {
+                    function.schema.Serialize(function.outputFile.FullName);
+                }
+                ret = functions.All(f => f.HasValue);
             }
 
-            var functions = GenerateFunctions(exportedTypes).ToList();
-            foreach (var function in functions.Where(f => f.HasValue && !f.Value.outputFile.Exists).Select(f => f.Value))
+            loadContext.Unload();
+            return ret;
+
+            Assembly ResolveAssembly(AssemblyLoadContext context, AssemblyName assemblyName)
             {
-                function.schema.Serialize(function.outputFile.FullName);
+                Assembly result = null;
+
+                // First, check the assembly root dir. Assemblies copied with the app always wins.
+                var path = Path.Combine(assemblyRoot, assemblyName.Name + ".dll");
+                if (File.Exists(path))
+                {
+                    result = context.LoadFromAssemblyPath(path);
+
+                    if (result != null)
+                    {
+                        return result;
+                    }
+                }
+
+                // Then, check if the assembly is already loaded into the default context.
+                // This is typically the case for framework assemblies like System.Private.Corelib.dll.
+                if (AssemblyLoadContext.Default.Assemblies.FirstOrDefault(a => a.GetName().Equals(assemblyName)) is Assembly existing)
+                {
+                    path = existing.Location;
```

Comment:
`Location` may not be safe in all scenarios. Likely not going to be an issue, at least in the current state, but could become one in the future. For safety, it would be good to validate this.

Reply:
Validate and throw an exception if it's `null` or doesn't exist on disk? Or validate and skip to the resolver fallback?

Reply:
If you're addressing the comment below and returning this assembly, you won't have to worry about the validation and fallback.